crates/uv/src/commands/project/init.rs

```diff
@@ -571,6 +608,102 @@ impl InitProjectKind {
 
         Ok(())
     }
+
+    #[allow(clippy::fn_params_excessive_bools)]
+    async fn init_script(
+        self,
+        script_path: &PathBuf,
+        python: Option<String>,
+        connectivity: Connectivity,
+        python_preference: PythonPreference,
+        python_downloads: PythonDownloads,
+        cache: &Cache,
+        printer: Printer,
+        no_workspace: bool,
+        no_readme: bool,
+        no_pin_python: bool,
+        package: bool,
+        native_tls: bool,
+    ) -> Result<()> {
+        if no_workspace {
+            warn_user_once!("`--no_workspace` is a no-op for Python scripts, which are standalone");
+        }
+        if no_readme {
+            warn_user_once!("`--no_readme` is a no-op for Python scripts, which are standalone");
+        }
+        if package {
+            warn_user_once!("`--package` is a no-op for Python scripts, which are standalone");
+        }
+        let client_builder = BaseClientBuilder::new()
+            .connectivity(connectivity)
+            .native_tls(native_tls);
+
+        let reporter = PythonDownloadReporter::single(printer);
+
+        if script_path.try_exists()? {
+            anyhow::bail!("Script already exists at {}", script_path.to_str().unwrap());
+        }
+
+        let requires_python = get_python_requirement_for_new_script(
+            &python,
+            no_pin_python,
+            python_preference,
+            python_downloads,
+            &client_builder,
+            cache,
+            &reporter,
+        )
+        .await?;
+
+        if let Some(path) = script_path.parent() {
+            fs_err::tokio::create_dir_all(path).await?;
+        }
+        Pep723Script::create_new_script(script_path, requires_python.specifiers()).await?;
+
+        Ok(())
+    }
+}
+
+pub(crate) async fn get_python_requirement_for_new_script(
+    python: &Option<String>,
```

Comment:
Same feedback as in the other PR around `&Option<T>`. In this case, best would be `Option<&str>`, since you can always go from `Option<&String>` to `Option<&str>`, but not the other way around.

Reply:
Done, thanks for the tip!